script.js

```diff
@@ -22,13 +23,109 @@ let timer;
+const preloader = document.querySelector('.preloader');
+
 
 startBtn.addEventListener('click', checkFormValue);
 checkBtn.addEventListener('click', checkResult);
 endBtn.addEventListener('click', switchSlide);
 restartBtn.addEventListener('click', switchSlide);
 
 
+function sendRequest(url) {
+  return new Promise((resolve,reject) => {
+    let request = new XMLHttpRequest();
+
+    request.open('GET', url);
+
+    request.responseType = 'json';
+
+    let error = new Error('Ошибка при получении вопросов');
+
+    request.onload = () => {
+      if (request.status >= 400) {
+        reject(error);
+      } else {
+        resolve(request.response);
+        preloader.classList.add('none');
+      }
+    };
+
+    request.onerror = () => reject(error);
+
+    request.send();
+  });
+}
+
+
+function processRequest() {
+  sendRequest(requestUrl)
+    .then(response => {
+      let result = response.results;
+
+      let requestQuestions = result.map(item => item.question);
+      requestCorrectAnswers = result.map(item => item.correct_answer);
+      let requestAllAnswers = result.map(item => [...item.incorrect_answers, item.correct_answer]);
+      let questionsText = document.querySelectorAll('.question__text');
+      let answersText = Array.from(questions).map(item => item.querySelectorAll('.answer__text'));
+
+      insertTextQuestions(questionsText, requestQuestions);
+      insertTextAnswers(answersText, requestAllAnswers);
+    })
+    .catch(err => {
+      console.log(err);
+      preloader.classList.add('none');
+    });
+}
+
+processRequest();
+
+
+function insertTextQuestions(defaultQuestions, requestQuestions) {
+    Array.from(defaultQuestions).forEach((item, index) => {
+      item.textContent = requestQuestions[index];
+    });
+  }
+
+
+function insertTextAnswers(defaultAnswers, requestAnswers) {
+    let a, b, c, d;
+    function getRandomAnswerNumber() {
+        a = getRandomNumber(4);
+
+        do {
+          b = getRandomNumber(4);
+        } while (b === a)
+
+        do {
+          c = getRandomNumber(4);
+        } while (c === b || c === a)
+
+        do {
+          d = getRandomNumber(4);
+        } while (d === b || d === a || d === c)
+    }
+
+    defaultAnswers.forEach((arr, index) => {
+        getRandomAnswerNumber();
+        arr[a].textContent = requestAnswers[index][0];
+        arr[b].textContent = requestAnswers[index][1];
+        arr[c].textContent = requestAnswers[index][2];
+        arr[d].textContent = requestAnswers[index][3];
+
+        arr[a].previousElementSibling.value = requestAnswers[index][0];
+        arr[b].previousElementSibling.value = requestAnswers[index][1];
+        arr[c].previousElementSibling.value = requestAnswers[index][2];
+        arr[d].previousElementSibling.value = requestAnswers[index][3];
+    });
+  }
+
+function makeQuestionsConteinerHeight(question) {
+  questionsContainer.style.height = question.offsetHeight + 'px';
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/voronkova-ab/-game-Who-want-to-be-a-millionaire-/pullRequest?prid=7740148)
```suggestion
  questionsContainer.style.height = question.offsetHeight + "px";
```